### PR TITLE
feat: implement LazyFrame.head for Dask (plus: run tpc-h q2 query with Dask present)

### PR DESCRIPTION
**Pull request by MarcoGorelli**
I also added `head` to make sure this notebook could run

drive-by:
- remove the `ibis-framework` install from other notebooks (as we'll no longer benchmark running pandas/dask via ibis anyway)
- specify minimum dask version in pyproject.toml, add a separate CI job for the minimum dask version

## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

File: .github/workflows/extremes.yml [hunk 1]
```diff
@@ -44,7 +44,32 @@ jobs:
...
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install uv (Unix)
+        run: curl -LsSf https://astral.sh/uv/install.sh | sh
+      - name: install-minimu-versions
```
Comment (FBruzzesi) [inline .github/workflows/extremes.yml:71]: Tiny typo

```suggestion
      - name: install-minimum-versions
```

File: narwhals/_dask/dataframe.py [hunk 2]
```diff
@@ -136,6 +140,11 @@ def with_row_index(self: Self, name: str) -> Self:
     def rename(self: Self, mapping: dict[str, str]) -> Self:
         return self._from_native_dataframe(self._native_dataframe.rename(columns=mapping))
 
+    def head(self: Self, n: int) -> Self:
+        return self._from_native_dataframe(
+            self._native_dataframe.head(n=n, compute=False)
```
Comment (FBruzzesi) [inline narwhals/_dask/dataframe.py:145]: Should we also pass `npartititions=-1` to parse all partitions if needed? I am not sure how this works: [docs](https://docs.dask.org/en/latest/generated/dask.dataframe.DataFrame.head.html)

**Review by FBruzzesi — COMMENTED** [review]
Left couple of comments, didn't check the notebook though 🙃

File: narwhals/_dask/dataframe.py (hunk 2 above)
Comment (MarcoGorelli) [reply narwhals/_dask/dataframe.py]: good points, thanks

**Comment by MarcoGorelli** [comment]
thanks for your review!

**Pull request merged**